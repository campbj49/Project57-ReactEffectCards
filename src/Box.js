import React from "react";
import "./Box.css";

/** Box: displays simple box shape
 * 
 * Props:
 * - backgroundColor: background-color of the box
 * - width: width of the box in pixels
 * - height: height of the box in pixels
 * - remove: callback function for the removal of the current box
 * 
 * State:
 * - none
 * 
 * App --> Carousel --> Box
 */

 function Box({id, backgroundColor, width, height,remove}) {
  console.log(height);

  return (
    <div 
    className="Box" 
    style={{"backgroundColor":backgroundColor,
                                  "width":width,
                                  "height":height}}>
      <p onClick={remove} id = {id}>X</p>
    </div>
  );
}

export default Box;
