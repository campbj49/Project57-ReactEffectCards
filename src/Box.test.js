import { render, fireEvent } from "@testing-library/react";
import Box from "./Box";

//smoke test to make sure it renders in the first place from smoke test example
it("renders without crashing", function() {
  render(<Box height={200} 
        width={200} 
        backgroundColor="red"/>);
});

// snapshot test from snapshot example
it("matches snapshot", function() {
  const {asFragment} = render(<Box height={200} 
    width={200} 
    backgroundColor="red"/>);
  expect(asFragment()).toMatchSnapshot();
});