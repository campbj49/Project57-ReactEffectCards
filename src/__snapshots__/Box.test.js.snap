// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    style="background-color: red; width: 200px; height: 200px;"
  >
    <p>
      X
    </p>
  </div>
</DocumentFragment>
`;
